src/library/treeitem.cpp

```diff
@@ -82,34 +82,22 @@ void TreeItem::initFeatureRecursively(LibraryFeature* pFeature) {
     }
 }
 
-TreeItem* TreeItem::appendChild(
-        std::unique_ptr<TreeItem> pChild) {
-    insertChild(m_children.size(), pChild.get()); // transfer ownership
-    return pChild.release();
-}
-
 TreeItem* TreeItem::appendChild(
         QString label,
         QVariant data) {
-    auto pNewChild = std::make_unique<TreeItem>(
+    std::unique_ptr<TreeItem> pNewChild = std::make_unique<TreeItem>(
             std::move(label),
             std::move(data));
-    return appendChild(std::move(pNewChild));
-}
-
-void TreeItem::removeChild(int row) {
-    DEBUG_ASSERT(row >= 0);
-    DEBUG_ASSERT(row < m_children.size());
-    delete m_children.takeAt(row);
+    TreeItem* pRet = pNewChild.get();
+    insertChild(m_children.size(), std::move(pNewChild));
+    return pRet;
```

Comment:
I have tried all sorts of solution but finally gave up. The issue is that QModelIndex takes a raw pointer internally nad you cannot convert a raw pointer back to a shared pointer. 
A entire smart pointer based solution requires a big refactoring that is out of scope of this PR.

Reply:
What QModelIndex do you mean exactly?

Reply:
Here: 
https://github.com/mixxxdj/mixxx/blob/eff27abd571ac3e1b273ad4aef6407eceee0cfee/src/library/autodj/autodjfeature.cpp#L292

Reply:
Mhmm ok, i see the problem. Can you add a comment that says the pointer that is being returned is non-owning and only valid for the lifetime duration of the `TreeItem`?

Reply:
I'm not quite sure if that is completely correct since the pointer would also be invalid if children get removed...

Reply:
I will add a comment. When the child is removed, the TreeItem is deleted and this pointer becomes dangling ...
dangerous .... This could be IMHO the source of the crash. In this PR I took care to delete those dangling pointers, whenever the tree is touched.